backend/vxlan/device.go

```diff
@@ -305,7 +305,10 @@ func setAddr4(link *netlink.Vxlan, ipn *net.IPNet) error {
 			return fmt.Errorf("failed to delete IPv4 addr %s from %s", addr.String(), link.Attrs().Name)
 		}
 	}
-
+	// Ensure that the device has a /32 address so that no broadcast routes are created.
+	// This IP is just used as a source address for host to workload traffic (so
+	// the return path for the traffic has a decent address to use as the destination)
+	ipn.Mask = net.CIDRMask(32, 32)
```

Comment:
really minor nit: may want this in the caller if want to keep `setAddr4` generic. Or if this is the only use case (which I believe it is), may as well only require caller to provide an`IP` rather than `IPNet`.